src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/LegacyPropagator.cs

```diff
@@ -0,0 +1,190 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net;
+using System.Collections.Generic;
+
+namespace System.Diagnostics
+{
+    internal class LegacyPropagator : TextMapPropagator
+    {
+        internal static TextMapPropagator Instance { get; } = new LegacyPropagator();
+
+        public override IReadOnlyCollection<string> Fields { get; } = new HashSet<string>() { TraceParent, RequestId, TraceState, Baggage, CorrelationContext };
+
+        public override void Inject(Activity activity, object carrier, PropagatorSetterCallback setter)
+        {
+            if (activity is null || setter is null)
```

Comment:
It's valid / expected for them to be null?  If yes, then they should be nullable.  If no, should we be throwing instead?

Reply:
I am trying to not throw in general in the propagator APIs. for example if the caller want to use Activity.Current without bothering to check before calling. 

I'll make it nullable.